Solutions

SergiyVS opened the pull request.

SergiyVS committed:
Solutions

maxim2310 commented on src/arrayMethodJoin.js:
```diff
@@ -5,7 +5,35 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinedStr = '';
+    let separatorInn = separator;
+    const arrInn = [...this];
+
+    if (separator === undefined) {
+      separatorInn = ',';
+    }
```
add default separator like this instead of if block
[].__proto__.join2 = function(separator = ',')

maxim2310 commented on src/arrayMethodJoin.js:
```diff
@@ -5,7 +5,35 @@
+      }
+    }
+
+    if (arrInn.length === 0) {
+      return '';
+    }
```
this case you can move to start before creating variable and use this.length

maxim2310 commented on src/arrayMethodJoin.js:
```diff
@@ -5,7 +5,35 @@
+      separatorInn = ',';
+    }
+
+    for (let i = 0; i < arrInn.length; i++) {
+      if (arrInn[i] === null || arrInn[i] === undefined) {
+        arrInn[i] = '';
+      }
+    }
```
this loop you can combine with next one

maxim2310 commented on src/arrayMethodJoin.js:
```diff
@@ -5,7 +5,35 @@
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinedStr = '';
+    let separatorInn = separator;
+    const arrInn = [...this];
```
make naming more meaningful

maxim2310 reviewed (changes requested):
Good job!!!

SergiyVS committed:
1. Default separator added
2. Cases with length 0 and 1 moved to start
3. Two loops combined
4.  Naming changed
